pandas/tests/series/test_analytics.py

```diff
@@ -873,52 +873,73 @@ def test_unique(self):
-        # check both int and object
-        for s in [Series([1, 2, 3, 3]), Series(['1', '2', '3', '3'])]:
-            expected = Series([False, False, False, True])
-            assert_series_equal(s.duplicated(), expected)
-            assert_series_equal(s.drop_duplicates(), s[~expected])
-            sc = s.copy()
-            sc.drop_duplicates(inplace=True)
-            assert_series_equal(sc, s[~expected])
-
-            expected = Series([False, False, True, False])
-            assert_series_equal(s.duplicated(keep='last'), expected)
-            assert_series_equal(s.drop_duplicates(keep='last'), s[~expected])
-            sc = s.copy()
-            sc.drop_duplicates(keep='last', inplace=True)
-            assert_series_equal(sc, s[~expected])
-
-            expected = Series([False, False, True, True])
-            assert_series_equal(s.duplicated(keep=False), expected)
-            assert_series_equal(s.drop_duplicates(keep=False), s[~expected])
-            sc = s.copy()
-            sc.drop_duplicates(keep=False, inplace=True)
-            assert_series_equal(sc, s[~expected])
-
-        for s in [Series([1, 2, 3, 5, 3, 2, 4]),
-                  Series(['1', '2', '3', '5', '3', '2', '4'])]:
-            expected = Series([False, False, False, False, True, True, False])
-            assert_series_equal(s.duplicated(), expected)
-            assert_series_equal(s.drop_duplicates(), s[~expected])
-            sc = s.copy()
-            sc.drop_duplicates(inplace=True)
-            assert_series_equal(sc, s[~expected])
-
-            expected = Series([False, True, True, False, False, False, False])
-            assert_series_equal(s.duplicated(keep='last'), expected)
-            assert_series_equal(s.drop_duplicates(keep='last'), s[~expected])
-            sc = s.copy()
-            sc.drop_duplicates(keep='last', inplace=True)
-            assert_series_equal(sc, s[~expected])
-
-            expected = Series([False, True, True, False, True, True, False])
-            assert_series_equal(s.duplicated(keep=False), expected)
-            assert_series_equal(s.drop_duplicates(keep=False), s[~expected])
-            sc = s.copy()
-            sc.drop_duplicates(keep=False, inplace=True)
-            assert_series_equal(sc, s[~expected])
+    @pytest.mark.parametrize(
+        "tc",
+        [
+            Series([1, 2, 3, 3], dtype=np.dtype('int_')),
+            Series([1, 2, 3, 5, 3, 2, 4], dtype=np.dtype('int_')),
+            Series([1, 2, 3, 3], dtype=np.dtype('uint')),
+            Series([1, 2, 3, 5, 3, 2, 4], dtype=np.dtype('uint')),
+            Series([1, 2, 3, 3], dtype=np.dtype('float_')),
+            Series([1, 2, 3, 5, 3, 2, 4], dtype=np.dtype('float_')),
+            Series([1, 2, 3, 3], dtype=np.dtype('unicode_')),
+            Series([1, 2, 3, 5, 3, 2, 4], dtype=np.dtype('unicode_')),
+            Series([True, False, False]),
+            Series([True, False, True, False])
+        ]
+    )
```

Comment:
I would pull the bool tests out and do them separately (as they are simpler and don't need as much checking);
also just in-ilne the expected results, don't use an if for that. the way to do this is to pass in things as a pair

e.g.
```
@pytest.mark.parametrize("tc1, tc2",
[
   (Series(1,2, 3, 3, dtype='int_'), Series([1, 2, 3, 5,3 ,2, 4])),
.....
])
def test_drop_duplicates(self, tc1, tc2)
```